examples/migration/recursive_fibonacci/task_emulation_layer.h

```diff
@@ -55,49 +55,98 @@ class base_task {
     void operator() () const {
         base_task* parent_snapshot = m_parent;
         const_cast<base_task*>(this)->execute();
-        if (m_parent && parent_snapshot == m_parent) {
-            m_parent->release();
+        if (m_parent && parent_snapshot == m_parent && m_child_counter == 0) {
+            if (m_parent->remove_reference() == 0) {
+                m_parent->operator()();
+                delete m_parent;
```

Comment:
Any chance, now that we have recycle_as_continuation, that a stack allocated task becomes a continuation?

Reply:
It definitely possible because we introduce 2 sets of APIs one for stack created task and another for allocated tasks.
Not sure if it is really a problem, however  it is also error prone as old Task API :)